pallets/marketplace/src/lib.rs

```diff
@@ -768,7 +771,49 @@ pub mod pallet {
                     <StoredJobStatus<T>>::remove(&job_id.0, &job_id.1);
                     <StoredJobRegistration<T>>::remove(&job_id.0, &job_id.1);
                 }
-                _ => Err(Error::<T>::JobRegistrationUnmodifiable)?,
+                JobStatus::Matched => {
+                    T::MarketplaceHooks::finalize_job(job_id, T::RewardManager::refund(job_id))?;
+
+                    // Remove matching data
+
+                    for (p, _) in <AssignedProcessors<T>>::iter_prefix(&job_id) {
+                        <StoredMatches<T>>::remove(&p, &job_id);
+                        <StoredStorageCapacity<T>>::remove(&p);
```

Comment:
You are removing the whole capacity tracking instead of increasing it, see other mutate operations on `StoredStorageCapacity`

Reply:
You are completely right, fixed in the latest commit.